Issue with 32-bit - cast to pointer from integer of different size

Below is the error stack from stage 'fetch':

cp Verify.pm blib/lib/Crypt/OpenSSL/Verify.pm
Running Mkbootstrap for Verify ()
chmod 644 "Verify.bs"
"/home/cpan/pit/64bit/perl-5.28.1/bin/perl" "-Iinc" -MExtUtils::Command::MM -e 'cp_nonempty' -- Verify.bs blib/arch/auto/Crypt/OpenSSL/Verify/Verify.bs 644
"/home/cpan/pit/64bit/perl-5.28.1/bin/perl" "-Iinc" "/home/cpan/pit/64bit/perl-5.28.1/lib/5.28.1/ExtUtils/xsubpp"  -typemap '/home/cpan/pit/64bit/perl-5.28.1/lib/5.28.1/ExtUtils/typemap' -typemap '/home/cpan/pit/64bit/conf/perl-5.28.1/.cpanplus/5.28.1/build/6iwVSwOHSM/Crypt-OpenSSL-Verify-0.17/typemap'  Verify.xs > Verify.xsc
mv Verify.xsc Verify.c
cc -c  -I/usr/include/openssl -fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -O3 -Wall -Werror -DOPENSSL_API_COMPAT=0x10100000L   -DVERSION=\"0.17\" -DXS_VERSION=\"0.17\" -fPIC "-I/home/cpan/pit/64bit/perl-5.28.1/lib/5.28.1/i686-linux-64int/CORE"   Verify.c
Verify.xs: In function Ã¢ÂÂXS_Crypt__OpenSSL__Verify_ctx_error_codeÃ¢ÂÂ:
Verify.xs:397:43: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
         RETVAL = X509_STORE_CTX_get_error((X509_STORE_CTX *) INT2PTR(UV, ctx));
                                           ^
cc1: all warnings being treated as errors
make: *** [Makefile:345: Verify.o] Error 1

--- a/Verify.xs
+++ b/Verify.xs
@@ -119,22 +119,22 @@ static SV *callback = (SV *) NULL;
 
 static int cb1(ok, ctx)
     int ok;
-    UV *ctx;
+    IV *ctx;
 {
     dSP;
     int count;
     int i;
 
     //printf("Callback pointer: %p\n", ctx);
-    //printf("Callback UL of pointer %lu\n", PTR2UV(ctx));
+    //printf("Callback INT of pointer %lu\n", (unsigned long) PTR2IV(ctx));
     ENTER;
     SAVETMPS;
 
     PUSHMARK(SP);
     EXTEND(SP, 2);
 
-    PUSHs(newSVuv(ok));                 // Pass ok as integer on the stack
-    PUSHs(newSVuv(PTR2UV(ctx)));        // Pass pointer address as integer
+    PUSHs(newSViv(ok));                 // Pass ok as integer on the stack
+    PUSHs(newSViv(PTR2IV(ctx)));        // Pass pointer address as integer
     PUTBACK;
 
     count = call_sv(callback, G_SCALAR);  // Call the verify_callback()
@@ -389,15 +389,15 @@ to the point address to be used
 =cut
 
 int ctx_error_code(ctx)
-    UV ctx;
+    IV ctx;
 
     PREINIT:
 
     CODE:
-        /* printf("ctx_error_code - UL holding pointer: %lu\n", ctx); */
-        /* printf("ctx_error_code - Pointer to ctx: %p\n", (void *) INT2PTR(UV , ctx)); */
+        /* printf("ctx_error_code - int holding pointer: %lu\n", (unsigned long) ctx); */
+        /* printf("ctx_error_code - Pointer to ctx: %p\n", (void *) INT2PTR(SV * , ctx)); */
 
-        RETVAL = X509_STORE_CTX_get_error((X509_STORE_CTX *) INT2PTR(UV, ctx));
+        RETVAL = X509_STORE_CTX_get_error((X509_STORE_CTX *) INT2PTR(SV *, ctx));
 
     OUTPUT:
 

